### ISSUE TITLE
Newly released OIDC auth feature (3.7.0) not working

### ISSUE THREAD

**Issue by tdevopsottawa** [hashicorp/terraform-provider-azurerm#16895]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.0

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

N/A (authentication)

### Terraform Configuration Files

```hcl
terraform {
  backend "azurerm" {
    container_name       = "tfstate"
    key                  = "kubeflow/nonprod/auth/terraform.tfstate"
    resource_group_name  = "pep-ecomm-terraform"
    storage_account_name = "pepecommterraform01"
    subscription_id      = "80306051-70ac-4ae0-95c7-5e797ba30a9c"
  }
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.7.0"
    }
  }
}

provider "azurerm" {
  use_oidc = true
  skip_provider_registration = true
  features {}
}

resource "azurerm_user_assigned_identity" "identity" {
  resource_group_name = var.az_resource_group_name
  location            = var.az_location
  name                = var.identity_name
  tags                = var.tags
}

And here's the GitHub actions:
name: OIDC Push

on: push

permissions:
  id-token: write
  contents: read

jobs:
  run-terragrunt:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository
        uses: actions/checkout@v3

      - name: Set up Python 3.9
        uses: actions/setup-python@v2
        with:
          python-version: 3.9

      - name: Install python dependencies
        run: |
          python3 -m pip install --upgrade pip
          pip install pathlib argparse

      - name: Setup Terraform
        uses: hashicorp/setup-terraform@v1
        with:
          terraform_version: 1.2.0
          terraform_wrapper: false

      - name: Azure Login
        uses: azure/login@v1
        with:
          client-id: ${{ secrets.AZURE_CLIENT_ID }}
          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}

      - name: Terraform Init
        run: terraform init
        working-directory: terraform/nonprod

(AZURE_CLIENT_ID, AZURE_TENANT_ID, and AZURE_SUBSCRIPTION_ID are defined as GitHub repository secrets.)
```

### Debug Output/Panic Output

```shell
(I'm running in GitHub actions, do I can't do this).
```

### Expected Behaviour

Terraform init should have worked when run as the GitHub action.

The OIDC connection itself from GitHub to Azure works - I added another workflow to test that simply ran a few Azure CLI commands and it worked correctly. So the problem is with Terraform/AzureRM and not with GitHub or Azure.

### Actual Behaviour

Initializing the backend...
╷
╷
│ Error: Error building ARM Config: Authenticating using the Azure CLI is only supported as a User (not a Service Principal).
│ 
│ To authenticate to Azure using a Service Principal, you can use the separate 'Authenticate using a Service Principal'
│ auth method - instructions for which can be found here: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret
│ 
│ Alternatively you can authenticate using the Azure CLI by using a User Account.
│ 
│ 

### Steps to Reproduce

1. Create GitHub OIDC connection to Azure
2. Create GitHub Action according to above text
3. Have it run terraform code in above text

### Important Factoids

Running as a GitHub Action

### References

This is a new feature that was just added in the 3.7.0 release today. [#16555](https://github.com/hashicorp/terraform-provider-azurerm/issues/16555).

The documentation is badly formatted and barely legible (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_oidc) but I followed it to the best of my ability, using a confirmed to be working OIDC connection, and adding the use_oidc = true line to my providers definition. If there's other necessary steps, they're not in the documentation.

**Comment by tombuildsstuff** [comment]
hi @tdevopsottawa 

OIDC needs to be configured for both [the Azure Backend](https://www.terraform.io/language/settings/backends/azurerm) and [the AzureRM Provider](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_oidc) - the config/error above is showing that you've only configured OIDC for the AzureRM Provider and not the backend, which is what's needed to fix this.

Thanks for the heads up about the docs, we'll get a PR to fix this shortly - but since this is a configuration question I'm going to close this issue, since this will be fixed by updating your configuration.

Thanks!

**Comment by tombuildsstuff** [comment]
Docs are fixed in #16896
